packages/kbn-apm-synthtrace-client/src/lib/apm/service.ts

```diff
@@ -20,19 +20,17 @@ export class Service extends Entity<ApmFields> {
   }
 }
 
-export function service(name: string, environment: string, agentName: string): Service;
-
-export function service(options: { name: string; environment: string; agentName: string }): Service;
-
-export function service(
-  ...args: [{ name: string; environment: string; agentName: string }] | [string, string, string]
-) {
-  const [serviceName, environment, agentName] =
-    args.length === 1 ? [args[0].name, args[0].environment, args[0].agentName] : args;
+export function service(options: {
+  name: string;
+  environment: string;
+  agentName: string;
+}): Service {
```

Comment:
why remove the other signature?

Reply:
I removed the other function overload to simply the implementation. This way if we need to pass additional parameters to the `service` function, we can pass it inside the object. With the other function overload where we were passing string, the params length would keep on increasing.

Also initially in this PR i was passing an additional parameter to this function for which i did this refactor. Do we need the other signature ?

Reply:
I prefer the string literal syntax. In fact I think we should default environment and agentName to `production` and `opbeans-java` (or something similar) so we can use `.service('my-service')`